core/src/main/java/hudson/model/Queue.java

```diff
@@ -1175,28 +1175,63 @@ public boolean contains(Task t) {
     /**
      * Checks if the given item should be prevented from entering into the {@link #buildables} state
      * and instead stay in the {@link #blockedProjects} state.
+     *
+     * @return the reason of blockage if it exists null otherwise.
      */
-    private boolean isBuildBlocked(Item i) {
-        if (i.task.isBuildBlocked() || !canRun(i.task.getResourceList()))
-            return true;
+    @CheckForNull
+    private CauseOfBlockage getCauseOfBlockageForItem(Item i) {
```

Comment:
sounds overused for non static method, why not just?
`getCauseOfBlockage(Item i)`
`getCauseOfBlockage(Task i)`

Reply:
This file contains already several `getCauseOfBlockage` methods, so, this name is just tradeoff to reduce cognitive complexity of this file :)

Reply:
not a problem while they have argument that are not inherited by each other =)